core/core-api/src/main/java/io/fabric8/launcher/core/spi/Application.java

```diff
@@ -23,8 +25,19 @@
 public @interface Application {
 
     enum ApplicationType {
+
         LAUNCHER,
-        OSIO
+        OSIO;
+
+        public static ApplicationType fromHeader(HttpServletRequest request) {
```

Comment:
may be beneficial to document public api

Reply:
Will do when we decide on https://github.com/fabric8-launcher/launcher-backend/pull/557#discussion_r225985580 with @gastaldi